packages/legend-application-query/src/stores/data-space/DataSpaceTemplateQueryCreatorStore.ts

```diff
@@ -93,21 +95,40 @@ export class DataSpaceTemplateQueryCreatorStore extends QueryEditorStore {
       this.dataSpacePath,
       this.graphManagerState.graph,
     );
-    const dataSpaceExecutableTemplate = guaranteeNonNullable(
-      dataSpace.executables
-        ?.filter(filterByType(DataSpaceExecutableTemplate))
-        .find((executable) => executable.id === this.templateQueryId),
-      `Can't find template query with id '${this.templateQueryId}'`,
+    let template = dataSpace.executables?.find(
+      (executable) => executable.id === this.templateQueryId,
     );
-    const executionContext = guaranteeNonNullable(
-      dataSpaceExecutableTemplate.executionContextKey
-        ? dataSpace.executionContexts.filter(
-            (ec) => ec.name === dataSpaceExecutableTemplate.executionContextKey,
-          )[0]
-        : dataSpace.defaultExecutionContext,
-      `Can't find execution context '${dataSpaceExecutableTemplate.executionContextKey}'`,
+    if (!template) {
+      template = dataSpace.executables?.find(
+        (executable) =>
+          executable instanceof DataSpacePackageableElementExecutable &&
+          executable.executable.value.path === this.templateQueryId,
+      );
+    }
+    if (!template) {
+      throw new IllegalStateError(
+        `Can't find template query with id '${this.templateQueryId}'`,
+      );
+    }
+    const executionContext = getExecutionContextFromDataspaceExecutable(
+      dataSpace,
+      template,
+    );
+    if (!executionContext) {
+      throw new IllegalStateError(
+        `Can't find a correpsonding execution context`,
+      );
+    }
+    const query = getQueryFromDataspaceExecutable(
+      template,
+      this.graphManagerState,
     );
-    this.templateQueryTitle = dataSpaceExecutableTemplate.title;
+    if (!query) {
+      throw new IllegalStateError(
```

Comment:
what is the error route look like here. Do we catch these errors ? 
When we throw this illegalStateerror does it cause a broken app ?

Reply:
This flow is for opening a template query in a new window. If there is an error, it will throw an error as below

https://github.com/user-attachments/assets/c18ea094-2878-4fa6-a87f-baa13b6a0094


e.g. 
http://localhost:9001/query/extensions/dataspace/org.finos.legend.query:legend-query-core-team:latest/domain::COVIDDatapace/template/template_id_1